stock_picking_actual_date/models/stock_picking.py

```diff
@@ -0,0 +1,36 @@
+# Copyright 2023 Quartile Limited
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    actual_date = fields.Date(help="Actual date of stock picking.")
+    is_editable_actual_date = fields.Boolean(
+        compute="_compute_is_editable_actual_date", string="Is Editable"
+    )
+
+    def _compute_is_editable_actual_date(self):
+        for rec in self:
+            rec.is_editable_actual_date = False
+            if rec.state not in ["done", "cancel"] or self.env.user.has_group(
+                "stock.group_stock_manager"
+            ):
+                rec.is_editable_actual_date = True
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "actual_date" in vals:
+            for rec in self:
+                if rec.state != "done":
+                    continue
+                account_moves = rec.move_ids.account_move_ids
+                if not account_moves:
+                    continue
+                account_moves.button_draft()
+                account_moves.name = False
+                account_moves.date = rec.actual_date
```

Comment:
I thought you told me that this would update the journal item amounts in foreign currency purchases according to the exchange rate. Is that not the case?

Reply:
No. It will not update. I called this [onchange](https://github.com/odoo/odoo/blob/b5a0560639efb6659762a22c7008d2f25bba1d8e/addons/account/models/account_move.py#L1586-L1589) method intentionally because I wanted to reflect exchange rate.